lib/OpenQA/WebAPI/Controller/API/V1/Bug.pm

```diff
@@ -0,0 +1,137 @@
+# Copyright (C) 2017 SUSE Linux LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package OpenQA::WebAPI::Controller::API::V1::Bug;
+use Mojo::Base 'Mojolicious::Controller';
+use OpenQA::IPC;
+use OpenQA::Utils;
+use OpenQA::Schema::Result::Jobs;
+use DBIx::Class::Timestamps 'now';
+use Date::Format 'time2str';
+use Try::Tiny;
+
+sub list {
+    my ($self) = @_;
+
+
+    my $bugs;
+    if ($self->param('refreshable')) {
+        my $delta = $self->param('delta') ? $self->param('delta') : 3600;
+        $bugs = $self->db->resultset("Bugs")->search(
+            {
+                -or => {
+                    is_refreshed => 0,
+                    t_updated    => {'<=' => time2str('%Y-%m-%d %H:%M:%S', time - $delta, 'UTC')}
+                },
+                is_existing => 1
+            });
+    }
+    else {
+        $bugs = $self->db->resultset("Bugs");
+    }
+    my $ret = {};
+
+    while (my $w = $bugs->next) {
+        next unless ($w->id);
+        $ret->{$w->id} = $w->bugid;
+    }
+    $self->render(json => {bugs => $ret});
+}
+
+
+sub show {
+    my ($self) = @_;
+
+    my $bug = $self->db->resultset("Bugs")->find($self->param('bugid'));
+
+    if ($bug) {
```

Comment:
better avoid the extra intendation by return after not found and rely on the $bug existance for the rest of the function